app/controllers/customers_controller.rb

```diff
@@ -326,11 +355,16 @@ def send_new_password(email)
     end
     begin
       newpass = String.random_string(6)
-      @customer.password = @customer.password_confirmation = newpass
+      if identity = @customer.identity
```

Comment:
What is the logic here?

Reply:
Please show all relevant lines of code. Just an if statement by itself is meaningless, I have to look up exactly where it is. This is another way of saying if bcrypted? do it the new way else bcrypt it. But this way uses only one db call instead of two.